app/models/answer.rb

```diff
@@ -41,5 +45,21 @@ def liked_by(uid)
   def comments
     self.drupal_comments
         .order('timestamp DESC')
+  end
+
+  def answer_notify(current_user)
```

Comment:
@jywarren Yes you are right. There is lot of repetitions. Inheritance is the best solution. I am looking this now. Solving it asap.